### ISSUE TITLE
A bug ?

### ISSUE THREAD

**Issue by Sererim** [corna/me_cleaner#414]
I was reading the me_cleaner.py and i think i found a bug
In RegionFile class
```py
 def read(self, n):
        if f.tell() + n <= self.region_end:
            return self.f.read(n)
        else:
            raise OutOfRegionException()

    def readinto(self, b):
        if f.tell() + len(b) <= self.region_end:
            return self.f.readinto(b)
        else:
            raise OutOfRegionException()
```

Why these methods work if f is undefined ?

I looked into it removed the argparser part and initilized the class and tried to use method read and pyton threw an error.

**Comment by nullifyism** [comment]
yeah buddy nice catch there matey, It does look like there's a potential issue with how " f " is referenced in the RegionFile class. The code you provided seems to lack a proper reference to the f attribute, which would likely cause an AttributeError. It should be self.f instead of just f, assuming f is meant to refer to an attribute of the class.

Here's how the corrected methods would look:

python
class RegionFile:
    def read(self, n):
        if self.f.tell() + n <= self.region_end:
            return self.f.read(n)
        else:
            raise OutOfRegionException()

    def readinto(self, b):
        if self.f.tell() + len(b) <= self.region_end:
            return self.f.readinto(b)
        else:
            raise OutOfRegionException()

Usin self.f ensures that the method refers to the instance's attribute and not some undefined variable. If self.f is not properly initialized in the class, that would also need to be addressed...
